tensorflow/cc/framework/gradients_test.cc

```diff
@@ -77,9 +106,10 @@ TEST_F(GradientsTest, OneMatMul) {
   for (const bool expected : {false, true}) {
     const Scope& scope = expected ? scope_expected_ : scope_test_;
     // Construct forward graph.
-    auto x = Const(scope, {{1.0, 2.0}, {3.0, 4.0}});
-    auto y = Const(scope, {{1.0, 0.0}, {0.0, 1.0}});
+    auto x = ThreeNodeVariable(scope, {{1.0, 2.0}, {3.0, 4.0}});
```

Comment:
You're right that differentiating w.r.t. Const nodes isn't realistic and prevented us from catching this error. However, it makes the unit tests simpler and easier to understand. The tests aren't worthless, they just test specific functionality, which is the idea of a unit test.

I think a good solution to the lack of coverage would be to have one or two "real world" tests that differentiate realistic graphs (we'll hopefully get these automatically when we switch the Python code to use this code under the hood). But I don't think we should add this much complexity to all of our existing unit tests in order to get slightly more coverage. For now, let's stick with the new tests you wrote that cover this particular case.